smdebug_rulesconfig/actions/actions.py

```diff
@@ -0,0 +1,184 @@
+from smdebug_rulesconfig.actions.utils import (
+    validate_training_job_prefix,
+    validate_email_address,
+    validate_phone_number,
+    validate_action_str,
+)
+
+
+class Action(object):
+    def __init__(self, **action_parameters):
+        """
+        Base class for action, which is to be invoked when a rule fires.. Offers `serialize` function to convert action
+        parameters to a string dictionary. This class is not meant to be initialized directly. Accepts dictionary of
+        action parameters and drops keys whose values are `None`.
+
+        :param action_parameters: Dictionary of action parameters.
+        """
+        action_parameters["name"] = self.__class__.__name__.lower()
+        self.action_parameters = {
+            key: value for key, value in action_parameters.items() if value is not None
+        }
+        validate_action_str(
+            self.serialize(), self.action_parameters
+        )  # sanity check, not expected to error!
+
+    def serialize(self):
+        """
+        Serialize the action parameters as a string dictionary.
+
+        :return: Action parameters serialized as a string dictionary.
+        """
+        return (
```

Comment:
Can we use the standard json library to handle serialization?

Reply:
No, see: https://github.com/awslabs/sagemaker-debugger-rulesconfig/pull/32#discussion_r545402799